### DISCUSSION TITLE
5.7.16: Unresolvable dependency

### DISCUSSION THREAD

**Discussion by m-lotze** [laravel/framework#26766] [General] [answered]
- Laravel Version: 5.7.16
- PHP Version: 7.2

### Description:
I have a project that produces this error after upgrading to 5.7.16 with composer update:

```
Loading composer repositories with package information
Updating dependencies (including require-dev)
Nothing to install or update
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover

In Container.php line 960:

  Unresolvable dependency resolving [Parameter #0 [ <required> $app ]] in class Illuminate\Support\Manager


Script @php artisan package:discover handling the post-autoload-dump event returned with error code 1
```

When i downgrade to 5.7.15 everything works. What could that be?

It seems to be only that project. Others work on 5.7.16.

**Comment by driesvints** [comment]
Can you first please try one of the following support channels? If you can actually identify this as a bug, feel free to report back.

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://webchat.freenode.net/?nick=laravelnewbie&channels=%23laravel&prompt=1)
- [StackOverflow](https://stackoverflow.com)

**Comment by mvanduijker** [comment]
I have the same problem with one of my projects...

**Comment by driesvints** [comment, marked as answer]
I just managed to update a project this way. This is probably related to something specific for your project. Perhaps an external package you use?

**Comment by m-lotze** [comment]
Found the problem:

I was missing `Illuminate\Notifications\NotificationServiceProvider::class,` in my app.php config file. Problaby because this project is quite old and in earlier versions this provider was not required there or i removed it.

**Comment by driesvints** [comment]
@m-lotze it was: https://laravel.com/docs/5.3/upgrade

**Comment by m-lotze** [comment]
Yep, saw that now. Edited my comment.

**Comment by devcircus** [comment]
Changes to laravel/laravel between versions, are documented in the upgrade guides.

**Comment by mvanduijker** [comment]
@m-lotze awesome! that fixed it for me!

**Comment by stevebauman** [comment]
This may be common knowledge but just in case someone runs into the same issue I had, you may need to delete your cached config manually after updating in `app/bootstrap/cache/config.php` as artisan broke when I didn't previously have the `NotificationServiceProvider` in my `config/app.php`.

**Comment by nowox** [comment]
I had the exact same issue: 
```
Unresolvable dependency resolving [Parameter #0 [ <required> $app ]] in class Illuminate\Support\Manager
```

I solved it just by changing the method prototype from this:

```php
public function __construct($app)
{
    $this->app = $app;
}
```

to this: 


```php
public function __construct(\Illuminate\Foundation\Application $app)
{
    $this->app = $app;
}
```

But in fact the origin was that I was missing the following provider:

```php
Illuminate\Notifications\NotificationServiceProvider::class
```

What remains mysterious to me is why my kludge was perfectly working and why do I need the NotificationServiceProvider if I am not using it?

**Comment by felixwetell** [comment]
Have had this issue a couple of times with different classes missing. This usually is solved by updating `composer`.
  
Run in the project folder: 
`composer update`

**Comment by naingminkhant** [comment]
Please Check your config/app.php

**Comment by rtconner** [comment]
I had to delete my `bootstrap/cache/*` files to get past this error

**Comment by arukompas** [comment]
Happened to me as well. Worked just fine for everyone on the team, but one member came across this issue. In the end we blamed the local git repository being corrupted. 

**We set up (cloned) the repo from scratch and then it was working just fine.**

Just something to try for anyone to whom other solutions didn't work.

**Comment by hamza-gaya** [comment]
> Can you first please try one of the following support channels? If you can actually identify this as a bug, feel free to report back.
> 
> * [Laracasts Forums](https://laracasts.com/discuss)
> * [Laravel.io Forums](https://laravel.io/forum)
> * [Discord](https://discordapp.com/invite/KxwQuKb)
> * [Larachat](https://larachat.co)
> * [IRC](https://webchat.freenode.net/?nick=laravelnewbie&channels=%23laravel&prompt=1)
> * [StackOverflow](https://stackoverflow.com)

dalal admi agar batana hai to batao nahi to warna chup raho

**Comment by hamza-gaya** [comment]
> I had the exact same issue:
> 
> ```
> Unresolvable dependency resolving [Parameter #0 [ <required> $app ]] in class Illuminate\Support\Manager
> ```
> 
> I solved it just by changing the method prototype from this:
> 
> ```
> public function __construct($app)
> {
>     $this->app = $app;
> }
> ```
> 
> to this:
> 
> ```
> public function __construct(\Illuminate\Foundation\Application $app)
> {
>     $this->app = $app;
> }
> ```
> 
> But in fact the origin was that I was missing the following provider:
> 
> ```
> Illuminate\Notifications\NotificationServiceProvider::class
> ```
> 
> What remains mysterious to me is why my kludge was perfectly working and why do I need the NotificationServiceProvider if I am not using it?

This is the perfect solution for this problems thanks for resolving it means a lot
